### PR TITLE
Added Python code in usaco-1018.mdx

### PR DESCRIPTION
**Pull request by 1a1242**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-1018.mdx [hunk 1]
```diff
@@ -84,5 +84,38 @@ int main() {
 }
 ```
 </CPPSection>
+<PySection>
 
+```python
+mod = 1000000007
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:90]: rename to MOD

File: solutions/usaco-1018.mdx [hunk 2]
```diff
@@ -84,5 +84,38 @@ int main() {
...
+
+#reading input from the file
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:92]:
```suggestion
```

File: solutions/usaco-1018.mdx [hunk 3]
```diff
@@ -84,5 +84,38 @@ int main() {
...
+filein = open('help.in', 'r') 
+n = int(filein.readline())
+
+#If the segment is open the value will be +ve else -ve
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:96]:
```suggestion
# If the segment is open the value will be +ve else -ve
```
same for other comments

File: solutions/usaco-1018.mdx [hunk 4]
```diff
@@ -84,5 +84,38 @@ int main() {
...
+a = [0] * (2 * n + 1)
+for i in range(n):
+	x, y = map(int, filein.readline().split())
+	a[x] += 1 #open
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:100]:
```suggestion
	a[x] += 1  # open
```
same for other comments

File: solutions/usaco-1018.mdx [hunk 5]
```diff
@@ -84,5 +84,38 @@ int main() {
...
+	pow[i] = pow[i - 1] * 2 % mod
+
+open_segs = 0 #number of open segments
+ans = 0
+for i in range(1, 2 * n + 1):
+	open_segs += a[i]
+	if a[i] == 1:
+		ans = (ans + pow[n - open_segs]) % mod
+
+#writing the output to the file
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:115]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1018.mdx [hunk 6]
```diff
@@ -84,5 +84,36 @@ int main() {
...
+
+# If the segment is open the value will be +ve else -ve
+a = [0] * (2 * n + 1)
+for i in range(n):
+	x, y = map(int, filein.readline().split())
+	a[x] += 1 # open
+	a[y] -= 1 # close
+	
+# precalculating the powers of 2
+pow = [1] * n
```
Comment (envyaims) [inline solutions/usaco-1018.mdx:103]: dont name it pow since that's an actual function name

File: solutions/usaco-1018.mdx [hunk 7]
```diff
@@ -84,5 +84,36 @@ int main() {
...
+for i in range(1,n):
+	pow[i] = pow[i - 1] * 2 % MOD
+
+open_segs = 0 # number of open segments
```
Comment (envyaims) [inline solutions/usaco-1018.mdx:107]:
```suggestion
open_segs = 0
```
extraneous comment

File: solutions/usaco-1018.mdx [hunk 8]
```diff
@@ -84,5 +84,36 @@ int main() {
...
 </CPPSection>
+<PySection>
 
+```python
+MOD = 1000000007
+
+filein = open('help.in', 'r') 
+n = int(filein.readline())
+
+# If the segment is open the value will be +ve else -ve
```
Comment (envyaims) [inline solutions/usaco-1018.mdx:95]: whats `ve`

**Review by envyaims — CHANGES_REQUESTED** [review]
tbh in python i dont think you even need to precalculate powers of 2 because it has built in modular exponentiation

**Comment by 1a1242** [comment]
The requested changes are done

File: solutions/usaco-1018.mdx [hunk 9]
```diff
@@ -84,5 +84,36 @@ int main() {
...
+
+open_segs = 0
+ans = 0
+for i in range(1, 2 * n + 1):
+	open_segs += a[i]
+	if a[i] == 1:
+		ans = (ans + power[n - open_segs]) % MOD
+
+fileout = open('help.out', 'w')
+fileout.write(str(ans) + '\n')
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:115]:
```suggestion
print(ans, file=open('help.out', 'w'))
```

File: solutions/usaco-1018.mdx [hunk 10]
```diff
@@ -84,5 +84,36 @@ int main() {
 }
 ```
 </CPPSection>
+<PySection>
 
+```python
+MOD = 1000000007
+
+filein = open('help.in', 'r') 
+n = int(filein.readline())
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:93]:
```suggestion
n = int(open('help.in'.readline())
```

File: solutions/usaco-1018.mdx [hunk 11]
```diff
@@ -84,5 +84,36 @@ int main() {
...
+
+# If the segment is open the value will be positive else negative
+a = [0] * (2 * n + 1)
+for i in range(n):
+	x, y = map(int, filein.readline().split())
+	a[x] += 1 # open
+	a[y] -= 1 # close
```
Comment (SansPapyrus683) [inline solutions/usaco-1018.mdx:100]:
```suggestion
	a[x] += 1  # open
	a[y] -= 1  # close
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1018.mdx (hunk 11 above)
Comment (1a1242) [reply solutions/usaco-1018.mdx]: Done with the changes

**Review by SansPapyrus683 — APPROVED** [review]
@envyaims

**Review by envyaims — APPROVED** [review]
okei

**Pull request merged**